content/manuals/build/building/best-practices.md

```diff
@@ -600,13 +600,30 @@ temporarily for a single `RUN` instruction, and don't persist in the final
 image. If you need to include files from the build context in the final image,
 use `COPY`.
 
+#### ADD or `curl`/`wget` and equivalents
+
 The `ADD` instruction is best for when you need to download a remote artifact
-as part of your build. `ADD` is better than manually adding files using
+as part of your build. 
+
+`ADD` is better than manually adding files using
 something like `wget` and `tar`, because it ensures a more precise build cache.
 `ADD` also has built-in support for checksum validation of the remote
 resources, and a protocol for parsing branches, tags, and subdirectories from
 [Git URLs](/reference/cli/docker/buildx/build.md#git-repositories).
 
+> [!NOTE]
+>
+> `ADD` redownloads the file every time the image is built to verify the checksum 
```

Comment:
This is not correct for servers that support etags. Only `HEAD` request is needed to know the file has not changed if there is local cache.

Reply:
Is this better?